product/views.py

```diff
@@ -1,3 +1,67 @@
-from django.shortcuts import render
+from django.views import View
+from django.http  import JsonResponse
+from django.db.models import Min, Q
 
-# Create your views here.
+from product.models import Product, ProductImage, ProductSize, Brand, BrandLine
+from bidding.models import Bidding, Status
+from account.models import User, Wishlist
+from account.utils  import login_check
+
+class ProductFilterView(View):
+    @login_check
+    def get(self, request):
+        try:
+            serch      = request.GET.get('q')
+            brand      = request.GET.get('brand')
+            brand_line = request.GET.get('brand_line')
+            subline    = request.GET.get('subline')
+            size       = request.GET.get('size')
+            sort       = request.GET.get('sort')
+            user_id    = request.user.id if request.user else None
+
+            PAGE   = int(request.GET.get('page', 0))
+            ITEM   = 20
+            OFFSET = ITEM * PAGE
+            LIMIT  = OFFSET + ITEM
+
+            filter = {}
+            if brand:
+                filter['brand'] = brand
+            if brand_line:
+                filter['brand_line'] = brand_line
+            if subline:
+                filter['subline'] = subline
+            if size:
+                filter['size__id'] = size
+            
+            serch_list = []
+            if serch:
+                serch_list.append(Q(korean_name__icontains=serch)|Q(english_name__icontains=serch)|Q(models_number__icontains=serch)|Q(brand__name__icontains=serch)|Q(brand_line__name__icontains=serch)|Q(sub_line__name__icontains=serch))
```

Comment:
```suggestion
            search_list = []
            if search:
                search_list.append(Q(korean_name__icontains=search)|Q(english_name__icontains=search)|Q(models_number__icontains=search)|Q(brand__name__icontains=search)|Q(brand_line__name__icontains=search)|Q(sub_line__name__icontains=search))
```
이렇게 길어지는경우 멀티라인으로 변경해서 가독성 좋게 만들어주세요!